CDEPS-interface/ufs/cdeps_share/shr_is_restart_fh_mod.F90

```diff
@@ -0,0 +1,146 @@
+module shr_is_restart_fh_mod
+
+  ! Common methods for components to check if it's time to write forecast hour-based restarts
+
+  !use dshr_methods_mod , only : chkerr
+  use ESMF, only : ESMF_ConfigCreate, ESMF_ConfigDestroy, ESMF_ConfigLoadFile, &
+                     ESMF_ConfigGetLen, ESMF_ConfigGetAttribute, ESMF_TimePrint, &
+                     ESMF_LOGMSG_INFO, ESMF_LogWrite, ESMF_TimeInterval, &
+                     ESMF_Time, ESMF_KIND_R8, ESMF_Config, ESMF_Clock, &
+                     ESMF_TimeIntervalSet, ESMF_TimePrint, operator(+), operator(==), &
+                     ESMF_LogFoundError, ESMF_LOGERR_PASSTHRU
+
+  implicit none
+  private
+  save
+
+  logical :: write_restartfh = .false.
+  type(ESMF_Time), allocatable :: restartFhTimes(:)
+!$OMP THREADPRIVATE (write_restartfh, restartFhTimes)
+
+  public :: init_is_restart_fh, is_restart_fh, finalize_restart_fh, write_restartfh
+
+contains
+
+  !-----------------------------------------------------------------------
+  subroutine init_is_restart_fh(currentTime, dtime, lLog)
+    !
+    ! !DESCRIPTION:
+    ! Process restart_fh attribute from model_configure in UFS
+    !
+    ! !USES:
+    !
+    ! !ARGUMENTS:
+    type(ESMF_Time), intent(in) :: currentTime
+    integer, intent(in)         :: dtime ! time step (s)
+    logical, intent(in)         :: lLog ! If true, this task logs restart_fh info
+    !
+    ! !LOCAL VARIABLES:
+    character(len=256)           :: timestr
+    integer                      :: n, nfh, fh_s, rc
+    logical                      :: isPresent
+    real(kind=ESMF_KIND_R8), allocatable :: restart_fh(:)
+    type(ESMF_TimeInterval)      :: fhInterval
+    type(ESMF_Config)            :: CF_mc
+    
+    character(len=*), parameter :: subname = 'init_restart_fh'
+    !-----------------------------------------------------------------------
+
+    ! set up Times to write non-interval restarts
+    inquire(FILE='model_configure', EXIST=isPresent)
+    if (isPresent) then !model_configure exists. this is ufs run
+      CF_mc = ESMF_ConfigCreate(rc=rc)
+      call ESMF_ConfigLoadFile(config=CF_mc,filename='model_configure' ,rc=rc)
+      !if (ChkErr(rc,__LINE__,__FILE__)) return
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
```

Comment:
Can you use the cleaner form ? 

if (ChkErr(rc,__LINE__,u_FILE_u)) return

Add

use dshr_methods_mod , only : chkerr

character(*), parameter :: u_FILE_u = &
       __FILE__

Reply:
We can, but that brings additional dependencies into this module besides ESMF. There are also quite some duplicate ChkErr functions floating around all just wrapping an ESMF function

Reply:
Marking as resolved, please let me know if this requires a new separate issue.